contracts/strategies/ltip-simple/LTIPSimpleStrategy.sol

```diff
@@ -0,0 +1,666 @@
+            Recipient storage recipient = _recipients[recipientId];
+
+            // only pending applications can be updated
+            // and the new status can only be Accepted or Rejected
+            if (
+                recipient.recipientStatus != Status.Pending
+                    || (recipientStatus != Status.Accepted && recipientStatus != Status.Rejected)
+            ) {
+                revert RECIPIENT_ERROR(recipientId);
+            }
+            recipient.recipientStatus = recipientStatus;
+
+            emit Reviewed(recipientId, recipientStatus, msg.sender);
+
+            unchecked {
+                ++i;
+            }
+        }
+    }
+
+    /// @notice Cancel (remove) recipient(s) application(s)
+    /// @dev You can remove multiple recipients at once or just one.
+    /// @param __recipientIds Ids of the recipients
+    function cancelRecipients(address[] calldata __recipientIds) external virtual onlyPoolManager(msg.sender) {
+        uint256 recipientLength = __recipientIds.length;
+
+        for (uint256 i; i < recipientLength;) {
+            address recipientId = __recipientIds[i];
+            Recipient storage recipient = _recipients[recipientId];
+
+            // if the status is none or appealed then revert
+            if (recipient.recipientStatus == Status.None || recipient.recipientStatus == Status.Canceled) {
+                revert RECIPIENT_ERROR(recipientId);
+            }
+
+            recipient.recipientStatus = Status.Canceled;
+
+            emit Canceled(recipientId, msg.sender);
+
+            unchecked {
+                ++i;
+            }
+        }
+    }
+
+    /// @notice Toggle the status between active and inactive.
+    /// @dev 'msg.sender' must be a pool manager to close the pool. Emits a 'PoolActive()' event.
+    /// @param _flag The flag to set the pool to active or inactive
+    function setPoolActive(bool _flag) external onlyPoolManager(msg.sender) {
+        _setPoolActive(_flag);
+    }
+
+    /// @notice Withdraw the tokens from the pool
+    /// @dev Callable by the pool manager
+    /// @param _token The token to withdraw
+    function withdraw(address _token) external virtual onlyPoolManager(msg.sender) {
+        uint256 amount = _getBalance(_token, address(this));
+
+        IAllo.Pool memory pool = allo.getPool(poolId);
+        _transferAmount(pool.token, msg.sender, amount);
+    }
+
+    /// @notice Sets the start and end dates.
+    /// @dev The timestamps are in seconds for the start and end times. The 'msg.sender' must be a pool manager.
+    ///      Emits a 'TimestampsUpdated()' event.
+    /// @param _registrationStartTime The start time for the registration
+    /// @param _registrationEndTime The end time for the registration
+    function updatePoolTimestamps(
+        uint64 _allocationStartTime,
+        uint64 _allocationEndTime,
+        uint64 _registrationStartTime,
+        uint64 _registrationEndTime
+    ) external onlyPoolManager(msg.sender) {
```

Comment:
^ would we need to update review and distribution timestamp ?

Reply:
Updated the timestamp setter to parse all timestamp and updated the `_isPoolTimestampValid` with a few lifecycle assumptions:

```
    function _isPoolTimestampValid(
        uint64 _allocationStartTime,
        uint64 _allocationEndTime,
        uint64 _distributionStartTime,
        uint64 _distributionEndTime,
        uint64 _registrationStartTime,
        uint64 _registrationEndTime,
        uint64 _reviewStartTime,
        uint64 _reviewEndTime
    ) internal view {
        if (
            // Register timestamps must be in the future
            block.timestamp > _registrationStartTime 
                // Start times must be before end times
                || _allocationStartTime > _allocationEndTime
                || _distributionStartTime > _distributionEndTime
                || _registrationStartTime > _registrationEndTime
                || _reviewStartTime > _reviewEndTime
                // Some end times must be after other end times
                || _registrationEndTime > _allocationEndTime
                || _allocationEndTime > _distributionStartTime
                || _reviewStartTime > allocationEndTime
        ) {
            revert INVALID();
        }
    }
    ```